### PR TITLE
added UI component to add Children's Card ( mockup based )

### PR DESCRIPTION
**Pull request by KostetskyiYaroslav**

### CODE REVIEW

**Comment by KostetskyiYaroslav** [comment]
image
image

File: MedicalDocManagment.WebUI/app/main/services/children-card.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    getChildrensCards(): ChildCardModel[]{
+        return this.childrenCardsList;
+    }
+
+    addChildrenCard(childrensCard: ChildCardModel): boolean {
+        if(childrensCard){
+            this.childrenCardsList.push(childrensCard);
+            return true;
+        } else {
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/services/children-card.service.ts]: remove else and invert condition

File: MedicalDocManagment.WebUI/app/main/services/diagnosis.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    getDiagnoses(): DiagnosisModel[]{
+        return this.diagnosesList;
+    }
+
+    addChildrenCard(diagnose: DiagnosisModel): boolean {
+        if(diagnose){
+            this.diagnosesList.push(diagnose);
+            return true;
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/services/diagnosis.service.ts]: remove else and invert condition

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+// import { NotificationsService, SimpleNotificationsComponent } from 'angular2-notifications';
+import DiagnosisModel from "../../models/diagnosis.model";
+import DiagnosisService from "../../services/diagnosis.service";
+
+@Component({
+    moduleId: module.id,
+    selector: 'children-card-add',
+    templateUrl: 'children-card-add.component.html',
+    providers: [ChildrensCardService, DiagnosisService]
+
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: empty line

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        this._childrensCardService.addChildrenCard(this.childrensCard);
+        console.log(this.childrensCard);
+        console.log(this._childrensCardService.getChildrensCards());
+        // this._notificationService.success("Успіх", "Дитячу картку успішно додано");
+    }
+
+    onDateChange(target: HTMLInputElement) {
+        let miliseconds: number = target.valueAsNumber;
+        this.childrensCard.date = (miliseconds) ? new Date(miliseconds) : null;
+        this.isDatesErrorsVisible = (this.childrensCard.date != null && this.dateFirstMoreSecond(this.limitDate, this.childrensCard.date))
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: create method

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            ? false
+            : true;
+    }
+
+    onCheckInChange(target: HTMLInputElement) {
+        let miliseconds: number = target.valueAsNumber;
+        this.childrensCard.checkIn = (miliseconds) ? new Date(miliseconds) : null;
+        this.isCheckInErrorsVisible = (this.childrensCard.checkIn != null && this.dateFirstMoreSecond(this.limitDate, this.childrensCard.checkIn))
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: create method

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            ? false
+            : true;
+    }
+
+    onCheckOutChange(target: HTMLInputElement) {
+        let miliseconds: number = target.valueAsNumber;
+        this.childrensCard.checkOut = (miliseconds) ? new Date(miliseconds) : null;
+        this.isCheckOutErrorsVisible = (this.childrensCard.checkOut != null && this.dateFirstMoreSecond(this.limitDate, this.childrensCard.checkOut))
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: create method

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        let miliseconds: number = target.valueAsNumber;
+        this.childrensCard.checkIn = (miliseconds) ? new Date(miliseconds) : null;
+        this.isCheckInErrorsVisible = (this.childrensCard.checkIn != null && this.dateFirstMoreSecond(this.limitDate, this.childrensCard.checkIn))
+            ? false
+            : true;
+    }
+
+    onCheckOutChange(target: HTMLInputElement) {
+        let miliseconds: number = target.valueAsNumber;
+        this.childrensCard.checkOut = (miliseconds) ? new Date(miliseconds) : null;
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: DRY out

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        this.isCheckOutErrorsVisible = (this.childrensCard.checkOut != null && this.dateFirstMoreSecond(this.limitDate, this.childrensCard.checkOut))
+            ? false
+            : true;
+    }
+
+    private dateFirstMoreSecond(f: Date, s: Date): boolean {
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts]: f & s bad names

File: MedicalDocManagment.WebUI/app/main/services/diagnosis.service.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import 'rxjs/add/observable/throw'
+
+import DiagnosisModel from '../models/diagnosis.model';
+
+// import { AuthenticationService } from '../login/authentication.service';
+
+@Injectable()
+export default class DiagnoseService {
+    private diagnosesList:DiagnosisModel[] = [
+  new DiagnosisModel({ id: 'A00.1', name: 'LABEL1'}),
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/services/diagnosis.service.ts]: format please

File: MedicalDocManagment.WebUI/app/main/models/child-card.model.ts [hunk 10]
```diff
@@ -18,10 +18,10 @@ export default class ChildCardModel {
             this.firstName = jsonObject.firstName;
             this.secondName = jsonObject.secondName;
             this.lastName = jsonObject.lastName;
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/main/models/child-card.model.ts:20]: create method

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    isDatesErrorsVisible = false;
+    isCheckInErrorsVisible = false;
+    isCheckOutErrorsVisible = false;
+    selectedClass = new ClassModel();
+    classes: ClassesModel;
+    selectedBlock = new BlockModel();
+    blocks: BlocksModel;
+    selectedNosology= new NosologyModel();
+    nosologies: NosologiesModel;
+    diagnoses: DiagnosesModel;
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts:34]: The preferred style for other properties is to initialize the field in the constructor (Google JavaScript Style Guide).

File: MedicalDocManagment.WebUI/app/main/main-app.component.ts [hunk 12]
```diff
@@ -3,7 +3,7 @@
 @Component({
     moduleId: module.id,
     selector: 'main-app-component',
-    templateUrl: 'main-app.component.html'
+    templateUrl: 'main-app.component.html',
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/app/main/main-app.component.ts]: Is this works with comma at the end of line?

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts [hunk 13]
```diff
@@ -0,0 +1,133 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+// import { NotificationsService, SimpleNotificationsComponent } from 'angular2-notifications';
+import ChildCardModel from '../../models/child-card.model';
+import DiagnosisModel from "../../models/diagnosis.model";
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts:5]: Prefer ' over " (Google JavaScript Style Guide).

**Review by AndriiKatsubo — CHANGES_REQUESTED** [review]

File: MedicalDocManagment.WebUI/app/main/main-app.component.ts [hunk 14]
```diff
@@ -3,7 +3,7 @@
...
 })
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/main/main-app.component.ts:7]: Is this works with comma at the end of line `templateUrl: 'main-app.component.html',`?

**Review by artempanko — APPROVED** [review]

**Review by Ravennod — APPROVED** [review]

File: MedicalDocManagment.WebUI/app/main/main-app.component.ts (hunk 14 above)
Comment (KostetskyiYaroslav) [reply MedicalDocManagment.WebUI/app/main/main-app.component.ts:7]: Of course yes 👍

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts (hunk 11 above)
Comment (KostetskyiYaroslav) [reply MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts:34]: Oh..sorry, I gonna do it next times

File: MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts (hunk 13 above)
Comment (KostetskyiYaroslav) [reply MedicalDocManagment.WebUI/app/main/children-cards-list/add/children-card-add.component.ts:5]: Oh, sorry, 
I gonna do it next times

**Pull request merged**